pkg/command/client.go

```diff
@@ -65,38 +72,41 @@ func (c *Client) SyncDashboard(ctx context.Context, uid string, queriesDir strin
 		if panel.Description == nil {
 			continue
 		}
-		queryName := ""
-		for _, part := range strings.Split(*panel.Description, "\n") {
-			queryParts := strings.Split(part, "query=")
-			if len(queryParts) != 2 {
-				continue
-			}
-			queryName = queryParts[1]
-		}
-		if queryName == "" {
+
+		targets := panel.GetTargets()
+		if targets == nil {
+			log.Printf("[%s:%s] panel has no targets found", panel.Type, panel.Title)
+			continue
+		} else if len(*targets) == 0 {
+			log.Printf("[%s:%s] panel has no targets found", panel.Type, panel.Title)
 			continue
 		}
 
-		var queryPath string
-		// support query name with and without .sql extension
-		if queryPath, _ = queries[queryName]; queryPath == "" {
-			if queryPath, _ = queries[fmt.Sprintf("%s.sql", queryName)]; queryPath == "" {
+		for i, part := range strings.Split(*panel.Description, "\n") {
+			queryName := ""
+			queryParts := strings.Split(part, "query=")
+			if len(queryParts) == 2 {
+				queryName = queryParts[1]
+			}
+
+			t := *targets
+			q := queries.Get(queryName)
+			if q == nil {
 				log.Printf("[%s:%s] query %s not found", panel.Type, panel.Title, queryName)
 				continue
 			}
-		}
-		queryBy, err := ioutil.ReadFile(queryPath)
-		if err != nil {
-			return err
-		}
 
-		// TODO(dm): support multiple targets?
-		targets := panel.GetTargets()
-		if targets != nil && len(*targets) > 0 {
-			t := *targets
-			t[0].RawSql = string(queryBy)
+			if i < len(t) {
```

Comment:
What would happen if the targets haven't yet been created in the dashboard panel?
Can we take the approach of overriding all the targets to match the queries defined in the panel description?

Eg; if the two queries were set in the panel description
* query=test-compliance/requests_num.promql
* query=test-compliance/requests_rate.promql

The panel would be updated to have two targets using the query/exp defined in the queries.

Reply:
rightnow if a panel only has one target, the second query will be ignored since we have only one target

Reply:
hmm, let me look into how to create target